HeapAttackIT » testTooManyEval

### CI Link

https://gradle-enterprise.elastic.co/s/uclhlvodhrova

### Repro line

./gradlew ':test:external-modules:test-esql-heap-attack:javaRestTest' --tests "org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT.testTooManyEval" -Dtests.seed=310090D8F9FDFC53 -Dtests.locale=sr-Latn -Dtests.timezone=Europe/Riga -Druntime.java=21

### Does it reproduce?

Didn't try

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

Probably related to https://github.com/elastic/elasticsearch/issues/104110 .

```
REPRODUCE WITH: ./gradlew ':test:external-modules:test-esql-heap-attack:javaRestTest' --tests "org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT.testTooManyEval" -Dtests.seed=310090D8F9FDFC53 -Dtests.locale=sr-Latn -Dtests.timezone=Europe/Riga -Druntime.java=21

org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT > testTooManyEval FAILED
    java.lang.AssertionError: Expected a map containing
       status: expected <429> but was <400>
        error: a map containing
           type: expected "circuit_breaking_exception" but was "parsing_exception"
```

More failures at:
* [elasticsearch / periodic / platform-support / main / oraclelinux-8 / platform-support-unix](https://gradle-enterprise.elastic.co/s/dryhguusqzs2k):
* [elasticsearch / periodic / platform-support / main / almalinux-8-aarch64 / checkPart1 / platform-support-arm](https://gradle-enterprise.elastic.co/s/glx65hkkg5fwu) where nodes failed. Might be transient, but mentioning it here if you'd like to take a look. "HeapAttack" sounds to me like something that could bring nodes down, that's why I'm mentioning it.

--- a/test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java
+++ b/test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java
@@ -280,10 +280,9 @@ public class HeapAttackIT extends ESRestTestCase {
         assertMap(map, matchesMap().entry("columns", columns).entry("values", hasSize(10_000)));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/104694")
     public void testTooManyEval() throws IOException {
         initManyLongs();
-        assertCircuitBreaks(() -> manyEval(1500));
+        assertCircuitBreaks(() -> manyEval(1000));
     }
 
     private Response manyEval(int evalLines) throws IOException {

